### ISSUE TITLE
Fails to build under Ruby 2.4.0.

### ISSUE THREAD

**Issue by bouchard** [backup/backup#820]
`backup` has a firm requirement on `json` version `1.8.2`, which fails to build with Ruby 2.4.0 due to `Fixnum` and `Bignum` being merged into `Integer`.

```shell
current directory: /usr/local/lib/ruby/gems/2.4.0/gems/json-1.8.2/ext/json/ext/generator
make "DESTDIR="
compiling generator.c
generator.c: In function ‘generate_json’:
generator.c:867:25: error: ‘rb_cFixnum’ undeclared (first use in this function)
     } else if (klass == rb_cFixnum) {
                         ^
generator.c:867:25: note: each undeclared identifier is reported only once for each function it appears in
generator.c:869:25: error: ‘rb_cBignum’ undeclared (first use in this function)
     } else if (klass == rb_cBignum) {
```

Updating to `json` version `2.0.2` corrects this problem. Please update the `gemspec` when you get a chance, thanks!

**Comment by nazarok** [comment]
+1

**Comment by tombruijn** [comment]
Looks like I can't update json to 2.0.2 because the qiniu gem has a lock on `~1.8`.
I'll report it upstream

**Comment by tombruijn** [comment]
I haven't been able to update the qiniu gem to support 2.4. Something breaks the entire thing under 2.4.

We'll wait to see if they can fix it, otherwise we'll have to remove the dependency and storage until they support it.
